### ISSUE TITLE
SyncBatchnorm for distributed training?

### ISSUE THREAD

**Issue by stdereka** [Jingkang50/OpenOOD#186]
Hello!

Many NN architectures in the OpenOOD 1.5 benchmark (including ResNets) have a BatchNorm normalization layer. It is recommended to use torch.nn.SyncBatchNorm in multi-GPU training with DDP for mean/std synchronization between processes: https://pytorch.org/docs/1.8.1/generated/torch.nn.SyncBatchNorm.html?highlight=syncbatchnorm

However, the current version of OpenOOD doesn't use SyncBatchNorm in DDP models, which may significantly affect multi-GPU results presented in the leaderboards.

**Comment by zjysteven** [comment]
Hi @stdereka, thank you for pointing this out. Yes this is some training detail that could be improved, although I'm a little doubtful whether it will "significantly" affect the results, as the [Pytorch official imagenet training example](https://github.com/pytorch/examples/blob/main/imagenet/main.py) doesn't use SyncBN.

We may rerun some experiments in the near future to see if this causes any significant issue. Meanwhile, you can also perform some training with SyncBN applied (potentially also the fix mentioned in #181) and compare the results with those on the current leaderboard. In fact we would greatly appreciate such efforts from the community to continuously improve OpenOOD!

**Comment by zjysteven** [comment]
We've added SyncBN in #223. We also reran a potentially affected method (ConfBranch) on ImageNet200. The results below show that the effect is negligible.

|             |       reported numbers (near-/far-OOD)     | numbers after rerunning (near-/far-OOD) |
|-------------|:---------------------------:|:-----------------------------------------------------------:|
| AUROC       | 79.10 ± 0.24 / 90.43 ± 0.18 |                 79.09 ± 0.06 / 90.43 ± 0.13                 |
| AUPR        | 82.11 ± 0.30 / 88.67 ± 0.27 |                 81.97 ± 0.03 / 88.60 ± 0.13                 |
| FPR@95      | 61.44 ± 0.34 / 34.75 ± 0.63 |                 61.16 ± 0.69 / 33.72 ± 0.31                 |
| ID Accuracy |         85.92 ± 0.07        |                         86.08 ± 0.07                        |
